core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -245,13 +245,18 @@ class BaseUnit : INamed, IConstruction {
             unitType.name -> true
             name -> true
             "All" -> true
+            
+            "Melee" -> unitType.isMelee()
+            "Ranged" -> unitType.isRanged()
```

Comment:
There is a Problem.
The Problem is, that both "Ranged" and "Melee" are both *attack types* and *unit types*. Meaning we need to decide, if something says that it affects Melee units, do we mean 'attack type melee' or 'unit type melee'?
Beforehand, it was 'unit type'. But I'm fine with attack type, as implemented here.
The real goal is to have unit types defined in a separate json, and then we can call them whatever we want, but that's a story for another time...
I'm thinking Archery and Gunpowder to replace Ranged, and Infantry to replace Melee, which differentiates them from Mounted units.

Reply:
All land ranged units in civ5 are archery (including bazooka and machine guns), and all land melee units starting with musketman are gunpowder (but tercio is still a melee).

Reply:
I have no problem dividing land melee into Infantry and Gunpowder, but Bazookas aren't Archery. That's silly.
This does mean we'll need to find a different name for them though.
Or perhaps not. Perhaps Type can *usually* determine Attack Type but can be overridden at the unit level. Food for thought.

Reply:
What about "Shooter"? Would that fit both Archers and Machine Guns?